src/storage/gc_worker.rs

```diff
@@ -0,0 +1,273 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::engine::{Engine, ScanMode, StatisticsSummary};
+use super::metrics::*;
+use super::mvcc::{MvccReader, MvccTxn, MAX_TXN_WRITE_SIZE};
+use super::txn::GC_BATCH_SIZE;
+use super::{Callback, Error, Key, Result};
+use kvproto::kvrpcpb::Context;
+use prometheus::local::{LocalHistogram, LocalIntCounter, LocalIntCounterVec};
+use std::fmt::{self, Display, Formatter};
+use std::sync::{Arc, Mutex};
+use util::worker::{self, Builder, Runnable, ScheduleError, Worker};
+
+pub const GC_MAX_PENDING: usize = 2;
+
+struct GCTask {
+    pub ctx: Context,
+    pub safe_point: u64,
+    pub callback: Callback<()>,
+}
+
+impl Display for GCTask {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let epoch = match self.ctx.region_epoch.as_ref() {
+            None => "None".to_string(),
+            Some(e) => format!("{{{}, {}}}", e.conf_ver, e.version),
```

Comment:
I think we can extract a utility function for formatting region epoch since elsewhere will use it too.

Reply:
I found it's better to use "{:?}" directly